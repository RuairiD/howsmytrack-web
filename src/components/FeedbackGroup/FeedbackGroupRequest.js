import React from "react";

import { Col, Divider, Row, Typography } from "antd";

import FeedbackRequestSummary from "../FeedbackRequestSummary/FeedbackRequestSummary";
import ReceivedFeedback from "./ReceivedFeedback";

const FeedbackGroupRequest = ({ feedbackRequestSummary, feedbackReceived }) => (
    <React.Fragment>
        <Divider />

        <Row gutter={[16, 16]}>
            <Col>
                <Typography.Title level={3}>Feedback for your submission</Typography.Title>
                <FeedbackRequestSummary
                    feedbackRequestSummary={feedbackRequestSummary}
                />
            </Col>
        </Row>

        <Row gutter={[16, 16]}>
            <Col>
                <ReceivedFeedback feedbackReceived={feedbackReceived} />
            </Col>
        </Row>
    </React.Fragment>
);

export default FeedbackGroupRequest;
