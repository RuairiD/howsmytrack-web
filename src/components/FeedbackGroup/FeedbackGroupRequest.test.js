import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackGroupRequest from "./FeedbackGroupRequest";

describe("FeedbackGroupRequest", () => {
    it("renders a FeedbackRequestSummary and ReceivedFeedback for the given request", () => {
        const wrapper = shallow(
            <FeedbackGroupRequest
                feedbackRequestSummary={{
                    feedbackRequestId: 1901,
                    mediaUrl: "mediaUrl",
                    mediaType: "mediaType",
                    feedbackPrompt: "feedbackPrompt",
                    emailWhenGrouped: false,
                    genre: "genre",
                }}
                feedbackReceived={[
                    {
                        feedbackResponseId: 1901,
                        feedback: "pretty good",
                        currentRating: 5,
                        allowReplies: false,
                        replies: 3,
                        unreadReplies: 2,
                    },
                ]}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
