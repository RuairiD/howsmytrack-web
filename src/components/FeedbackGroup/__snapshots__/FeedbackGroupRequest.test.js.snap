// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackGroupRequest renders a FeedbackRequestSummary and ReceivedFeedback for the given request 1`] = `
<Fragment>
  <Divider />
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Title
        level={3}
      >
        Feedback for your submission
      </Title>
      <FeedbackRequestSummary
        feedbackRequestSummary={
          Object {
            "emailWhenGrouped": false,
            "feedbackPrompt": "feedbackPrompt",
            "feedbackRequestId": 1901,
            "genre": "genre",
            "mediaType": "mediaType",
            "mediaUrl": "mediaUrl",
          }
        }
      />
    </Col>
  </Row>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <ReceivedFeedback
        feedbackReceived={
          Array [
            Object {
              "allowReplies": false,
              "currentRating": 5,
              "feedback": "pretty good",
              "feedbackResponseId": 1901,
              "replies": 3,
              "unreadReplies": 2,
            },
          ]
        }
      />
    </Col>
  </Row>
</Fragment>
`;
